supervisor/api/network.py

```diff
@@ -85,10 +86,10 @@
 )
 
 
-def ipconfig_struct(config: IpConfig) -> dict[str, Any]:
+def ipconfig_struct(config: IpConfig, setting: IpSetting) -> dict[str, Any]:
     """Return a dict with information about ip configuration."""
     return {
-        ATTR_METHOD: config.method,
+        ATTR_METHOD: setting.method,
```

Comment:
Don't we use this API to show the current status of the network device in the UI? If so I think this risks breaking those screens. Because if the network is down then it'll show what NM wants it to be based on the setting rather then what it actually is right now won't it?

Reply:
There is no "current method". The method got always copied over in the `from_dbus_interface()` function. This just moves that copy closer to the API.

Reply:
I thought the active connection could show up as disabled when it failed to initialize even if the setting said auto or static. This was actually a bug we had right? Where we would accidentally disable the interface permanently at startup if there was a connection error by writing the value in `ActiveConnection` into persistent settings.

I agree with the separation I just want to make sure that we don't break the frontend. If our network status screen showed "disabled" for method due to a network initialization error even if the method was "auto" in settings, it should keep doing that.

Reply:
ah right, good point, ok same behavior then